### DISCUSSION TITLE
Questions about SOAP descriptors

### DISCUSSION THREAD

**Discussion by rajathsalegame** [SINGROUP/dscribe#60] [Q&A] [answered]
Hi, I had a few questions about the SOAP implementation within `dscribe` that I was hoping someone could answer:

1) What are some empirical ways to make sure that the SOAP outputs I am seeing are sensible and make sense (either visual or mathematical?
2) What are some strategies to decide on the best hyperparameters for SOAP generation (setting `r_cut`, `n_max`, `l_max`)?
3) Just making sure, but if I specify the positions to calculate as `[1,4,10]` for a given molecule, the SOAP output will be in the order `[SOAP_VECTOR[1], SOAP_VECTOR[4], SOAP_VECTOR[10]]`, correct?

Thanks!

**Comment by lauri-codes** [comment, marked as answer]
Hi @rajathsalegame.

1. With SOAP this is quite hard as the features are so abstract. But here are a few tips that come to mind:
   - You can plot the flattened features and color/label different parts based on the ranges that you can get out of soap.get_location(("H", "O"))
   - You can calculate the dot product of normalized SOAP feature vectors to get a rough idea of the similarity of different outputs. The values will be between [0, 1], where 0 means completely different, 1 is completely identical.
2. In a supervised machine learning application, you would include these as hyperparameters to your cross-validation. This is a very standard way to select good ones. For unsupervised learning or other uses, the selection becomes much harder. The pragmatic way would be to test a few settings or do a literature search for values others have used.
3. You are correct.

**Reply by rajathsalegame** [reply]
Thank you!
